html/ajax_search.php

```diff
@@ -20,6 +20,13 @@
     if (strlen($search) > 0) {
         $found = 0;
 
+        if (!Auth::user()->hasGlobalRead()) {
+            $device_ids = Permissions::devicesForUser()->implode(',');
+            $perms_sql = "`D`.`device_id` IN ($device_ids)";
```

Comment:
This one't work you need to have placeholders...
dbGenPlaceholders() I think

Reply:
It should work anyway, it just generates SQL like `IN (1,2,3)`.  
Placeholders are kinda there for SQL injections, which won't happen since it's not user input, so I thought it was a fair trade off.  
@murrant If it's a deal breaker for you, let me know.

Reply:
@Jellyfrog but then the code will escape "(1,2,3)" (to avoid sql injection)

Reply:
it kinda is, no reason to leave the query vulnerable to sql injection (no matter how unlikely)